com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/NewObjectManipulator.cs

```diff
@@ -0,0 +1,575 @@
+        /// </remarks>
+        protected virtual InteractionFlags CurrentInteractionType
+        {
+            get
+            {
+                if (IsGrabSelected)
+                {
+                    return InteractionFlags.Near;
+                }
+                else if (IsRaySelected)
+                {
+                    return InteractionFlags.Ray;
+                }
+                else if (IsGazePinchSelected)
+                {
+                    return InteractionFlags.Gaze;
+                }
+                else
+                {
+                    return InteractionFlags.Generic;
+                }
+            }
+        }
+
+        protected MoveTransformation moveManipulation;
+
+        protected RotateTransformation rotateManipulation;
+
+        protected ScaleTransformation scaleManipulation;
+
+        #endregion Protected Properties
+
+        #region Private Properties
+
+        private PlacementHub placementHub;
+
+        private bool ShouldSmooth => (IsGrabSelected && SmoothingNear) || (!IsGrabSelected && SmoothingFar);
+
+        private bool wasSmoothed;
+
+        private Rigidbody rigidBody;
+
+        private bool wasGravity = false;
+
+        private bool wasKinematic = false;
+
+        // Reusable list for fetching interactionPoints from interactors.
+        private List<Pose> interactionPoints = new List<Pose>();
+
+        // Reusable list for fetching attachPoints from interactors.
+        private List<Pose> attachPoints = new List<Pose>();
+
+        // Reusable list for fetching grabPoints from interactors.
+        private List<Pose> grabPoints = new List<Pose>();
+
+        #endregion Private Properties
+
+        #region MonoBehaviour Functions
+
+        protected virtual void ApplyRequiredSettings()
+        {
+            // ObjectManipulator is never selected by poking.
+            DisableInteractorType(typeof(IPokeInteractor));
+        }
+
+        protected override void Reset()
+        {
+            base.Reset();
+            ApplyRequiredSettings();
+            selectMode = InteractableSelectMode.Multiple;
+        }
+
+        private void OnValidate()
+        {
+            ApplyRequiredSettings();
+        }
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            placementHub = transform.GetComponent<PlacementHub>();
+
+            ApplyRequiredSettings();
+
+            rigidBody = GetComponent<Rigidbody>();
+
+            InstantiateManipulationTransformations();
+        }
+
+        #endregion
+
+        private void InstantiateManipulationTransformations()
+        {
+            moveManipulation = new MoveTransformation(this);
+            rotateManipulation = new RotateTransformation(this);
+            scaleManipulation = new ScaleTransformation(this);
+
+            moveManipulation.logic = Activator.CreateInstance(ManipulationLogicTypes.moveLogicType) as ManipulationLogic<Vector3>;
```

Comment:
Are these `MoveLogic`s going to be Transformations at some point?

Reply:
I think for sake of back compatibility we shouldn't remove the MoveLogics code into the Transformations class, at best we duplicate the code, but I'm a bit unsure of the best way to handle using a new interface while the code we need is still embedded in the older API.

@keveleigh for some perspective?